First draft of Azure Pipelines for the DevOps docs (#4766)
umbraco/UmbracoDocs: Tutorials/DevOps/Building-with-Azure-DevOps/index.md

    platform: '$(buildPlatform)'
    configuration: '$(buildConfiguration)'

- task: PublishBuildArtifacts@1
  displayName: 'Publish Artifacts'
  inputs:
    PathtoPublish: '$(Build.ArtifactStagingDirectory)'
    ArtifactName: 'drop'
    publishLocation: 'Container'

```

##What does this do?

The first block determines how this pipeline will be triggered we have specified two branches for Azure DevOps to listen for commits.

Pool is stating what OS the build agent should run. `windows-latest` or `ubuntu-latest` should be fine for most use cases. You can find a list of supported images [here](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software).

Variables is a section to define any sort of object such as a string or boolean to be used in your YAML.

Steps is used define what tasks need to happen in sequence to complete our build

Since this is a basic example we are just using Steps. You can incorporate Stages with steps to have a more complex build process. You can read more about Stages [here](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/stages?view=azure-devops&tabs=yaml)

> In our Steps we are telling Azure DevOps to restore our nuget packages, run VS Build and then package our compiled website into an artifact called drop.zip.

Reviewer: ```suggestion
In the Steps block, we are restoring the NuGet packages, running VS build, and then packaging the compiled website into an artifact called drop.zip.
```

> It does not matter if you use VS Build instead of Dot Net Core CLI when building Umbraco 9+ as they ultimately do the same thing. You will run into issues using the Dot Net Core CLI task for a .Net Framework project, Umbraco 8 and below.

Reviewer: ```suggestion
It does not matter if you use VS Build instead of .Net Core CLI when building Umbraco 9+ as they do the same thing. You will run into issues using the .Net Core CLI task for a .Net Framework project for Umbraco 8 and below.
```

> The above VS Build example includes a list of MS Build parameters to build and publish our website into a zip file to be passed on to the artifacts task.

Reviewer: ```suggestion
The VS Build example includes a list of MS Build parameters to build and publish the website into a zip file. This file is then passed on to the artifacts task.
```
